src/lib/openpath/queries.js

```diff
@@ -9,6 +10,12 @@ export const queryLastServiceAccount = async client => {
   return account && account.data?.length > 0 ? account.data[0] : null
 }
 
+export const queryAccountByToken = async (client, token) => {
+  const accountsQuery = buildAccountByToken({ token }).definition
+  const account = await client.query(accountsQuery)
+  return account && account.data?.length > 0 ? account.data[0] : null
```

Comment:
nit: No need to change, but I think `account?.data?.[0] || null` would be equivalent (or even better `account?.data?.[0] ?? null`

Reply:
True, must I must admit I'm not a big fan of this syntax, I find it confusing (but probably because I'm not used to it)